core/src/cluster_slots_service.rs

```diff
@@ -120,13 +130,23 @@ impl ClusterSlotsService {
     fn update_completed_slots(
         completed_slots_receiver: &CompletedSlotsReceiver,
         cluster_info: &ClusterInfo,
+        rpc_subscriptions: &Option<Arc<RpcSubscriptions>>,
     ) {
         let mut slots: Vec<Slot> = vec![];
         while let Ok(mut more) = completed_slots_receiver.try_recv() {
             slots.append(&mut more);
         }
         #[allow(clippy::stable_sort_primitive)]
         slots.sort();
+        if let Some(rpc_subscriptions) = rpc_subscriptions {
```

Comment:
This is run in a loop() with a 200ms sleep, so the responses may be a little delayed. Any reason not to plumb it directly into the blockstore insert?

Reply:
No particular reason, I'll move into blockstore

Reply:
hmmm looking at it now the `push_epoch_slots()` may actually be slow sometimes due to some 

1) Contention on the cluster_info lock in `self.time_gossip_read_lock` (this has been known to spike)
2) Serializing/deserializing epoch slots

It's kind of ugly, but we could plumb another thread in between here that reads from completed_slots_receiver and does notification...

Reply:
Done!